### DISCUSSION TITLE
Conditional Function for Screening Two Trials

### DISCUSSION THREAD

**Discussion by acer-helia** [jspsych/jsPsych#1930] [Q&A] [answered]
Hello all, I have a task where participants only move to the main set of trials if they get at least 1 of 2 of the screener trials correct. I am trying to write a conditional function for this, but no matter how I try to debug it I can either get it to always go to main trial no matter the response, or the opposite (i.e., no one moves to the main trials no matter the response). 

I've not used conditionals for more than one trial before, and have tried using the suggestions provided [here](https://github.com/jspsych/jsPsych/discussions/1758) and [here](https://github.com/jspsych/jsPsych/discussions/748) to no avail. 

I've copied a snippet of my code below. Any help is greatly appreciated!

```javascript
var voo_practice_b = {
        type: 'html-button-response',
        choices: ['Teacher','Minister','Lamp','Poet'],
        prompt: '<p style="text-align: centre">'+
            'Which word is the odd one out?<br><br> Please <b>click</b> on the word you think is correct.<br> If you are not sure, have a guess.',
        stimulus: "", data: {test_part: 'VOO Screen', target: 'Lamp', correct_response: 2},
        required: true,
        response_ends_trial: true,
        on_finish: function(data){
          if(jsPsych.pluginAPI.compareKeys(data.button_pressed, 2)){
            data.correct = true;
          }
          else {
            data.correct = false;
          }
        }

    var if_ab_errors = {
    	timeline: [voo_main], //should be displayed if at least 1 of 2 screens is correct
    	conditional_function: function(){
  		var counter = jsPsych.data.get().filter({test_part: 'VOO Screen', correct: true}).count(); 
    		if (counter => 1){
    			return true;
    			} else {
    			return false;
    			}
    		}
        };
```

**Comment by acer-helia** [comment, marked as answer]
Hi all - just an update on this, I was able to solve it! With jsPsych 6.3.1 button presses are now coded as data.response rather than data.button_pressed. Additionally, in my conditional function I needed to have separate filters for the filter tag for the trials I was using to screen for responses ({test_part: 'VOO Screen'}) and then for further counting the number of correct responses ({correct: true}).

I have copied my code below in case anyone else has a similar issue.

```
  var voo_practice_b = {
        type: 'html-button-response',
        choices: ['Teacher','Minister','Lamp','Poet'],
        prompt: '<p style="text-align: centre">'+
            'Which word is the odd one out?<br><br> Please <b>click</b> on the word you think is correct.<br> If you are not sure, have a guess.',
        stimulus: "", data: {test_part: 'VOO Screen', target: 'Lamp', correct_response: 2},
        required: true,
        response_ends_trial: true,
        on_finish: function(data){
          if(jsPsych.pluginAPI.compareKeys(data.response, 2)){   //use response instead of button_pressed
            data.correct = true;
          }
          else {
            data.correct = false;
          }
        }
      };

   var if_ab_errors = {
    timeline: [voo_main], //should be displayed if at least 1 of 2 screens is correct
    conditional_function: function(){
    var counter = jsPsych.data.get().filter({test_part: 'VOO Screen'}).last(2).filter({correct: true}).count(); //count if last two trials in tagged trials are correct, need to ensure test_part and correct are in separate filters
         if (counter > 0){
         return true;
         } else {
         return false;
        }
      }
    };
```

**Reply by becky-gilbert** [reply]
Hi Andrea, I was just going to reply to you but you beat me to it. Glad you figured it out!! 👍 

Since you've included the correct response in the trial data, one tiny suggestion is to compare the response against the correct response from the trial data when you're scoring the response in your `on_finish` function, like this:
```js
if(jsPsych.pluginAPI.compareKeys(data.response, data.correct_response)){
```
That way you won't have to manually change the correct button number in the `on_finish` function to get the scoring to work.

**Reply by becky-gilbert** [reply]
Hi again, I also just noticed that you don't need to use `compareKeys` at all here, because you're comparing integers rather than keys! Oops! So you can just do this:
```js
if(data.response == data.correct_response){
```

**Reply by acer-helia** [reply]
Thanks for the helpful tips Becky, definitely useful for future reference! 😃
